src/sdk/pynni/nni/nas/tensorflow/classic_nas/mutator.py

```diff
@@ -0,0 +1,215 @@
+                                                                self._search_space[mutable.key]["_value"])
+            elif isinstance(mutable, InputChoice):
+                result[mutable.key] = self._sample_input_choice(mutable, data["_idx"], data["_value"],
+                                                                self._search_space[mutable.key]["_value"])
+            elif isinstance(mutable, MutableScope):
+                logger.info("Mutable scope '%s' is skipped during parsing choices.", mutable.key)
+            else:
+                raise TypeError("Unsupported mutable type: '%s'." % type(mutable))
+        return result
+
+    def _standalone_generate_chosen(self):
+        """
+        Generate the chosen architecture for standalone mode,
+        i.e., choose the first one(s) for LayerChoice and InputChoice.
+        ::
+            { key_name: {"_value": "conv1",
+                         "_idx": 0} }
+            { key_name: {"_value": ["in1"],
+                         "_idx": [0]} }
+        Returns
+        -------
+        dict
+            the chosen architecture
+        """
+        chosen_arch = {}
+        for key, val in self._search_space.items():
+            if val["_type"] == LAYER_CHOICE:
+                choices = val["_value"]
+                chosen_arch[key] = {"_value": choices[0], "_idx": 0}
+            elif val["_type"] == INPUT_CHOICE:
+                choices = val["_value"]["candidates"]
+                n_chosen = val["_value"]["n_chosen"]
+                if n_chosen is None:
+                    n_chosen = len(choices)
+                chosen_arch[key] = {"_value": choices[:n_chosen], "_idx": list(range(n_chosen))}
+            else:
+                raise ValueError("Unknown key '%s' and value '%s'." % (key, val))
+        return chosen_arch
+
+    def _generate_search_space(self):
+        """
+        Generate search space from mutables.
+        Here is the search space format:
+        ::
+            { key_name: {"_type": "layer_choice",
+                         "_value": ["conv1", "conv2"]} }
+            { key_name: {"_type": "input_choice",
+                         "_value": {"candidates": ["in1", "in2"],
+                                    "n_chosen": 1}} }
+        Returns
+        -------
+        dict
+            the generated search space
+        """
+        search_space = {}
+        for mutable in self.mutables:
+            # for now we only generate flattened search space
+            if isinstance(mutable, LayerChoice):
+                key = mutable.key
+                val = mutable.names
+                search_space[key] = {"_type": LAYER_CHOICE, "_value": val}
+            elif isinstance(mutable, InputChoice):
+                key = mutable.key
+                search_space[key] = {"_type": INPUT_CHOICE,
+                                     "_value": {"candidates": mutable.choose_from, #["input_"+str(i) for i in range(mutable.n_candidates)],
```

Comment:
Please remove useless comments. Or make it explicitly useful.

Reply:
sure, forgot to remove